### PR TITLE
crud.rs: Introduce Query structure.

### PR DESCRIPTION
**Pull request by BearLemma**
The Query structure is used as an intermediary between raw URL and
QueryPlan.

### CODE REVIEW

File: server/src/datastore/crud.rs [hunk 1]
```diff
@@ -42,50 +42,12 @@ fn make_stream(
     tr: TransactionStatic,
 ) -> Result<impl Stream<Item = Result<JsonObject>>> {
     let url = Url::parse(&params.url)
-        .with_context(|| format!("failed to parse query string '{}'", params.url))?;
-    let query_plan = query_plan_from_url(context, &params.type_name, &url)?;
+        .with_context(|| format!("crud endpoint failed to parse url: '{}'", params.url))?;
```
Comment (espindola) [inline server/src/datastore/crud.rs:45]: The error message change is not related to introducing the Query structure.

File: server/src/datastore/crud.rs [hunk 2]
```diff
@@ -107,6 +69,76 @@ pub(crate) fn delete_from_url(c: &RequestContext, type_name: &str, url: &str) ->
...
+    /// Parses provided `url` and builds a `Query` that can be used to build a `QueryPlan`.
+    fn from_url(c: &'a RequestContext, entity_name: &str, url: &Url) -> Result<Self> {
+        let base_type = &c
+            .ts
+            .lookup_object_type(entity_name, &c.api_version)
+            .context("unexpected type name as crud query base type")?;
+
+        let mut q = Query::new(c, base_type.clone());
+        for (param_key, value) in url.query_pairs().into_owned() {
+            let op = parse_query_parameter(&base_type, &param_key, &value).with_context(|| {
```
Comment (espindola) [inline server/src/datastore/crud.rs:104]: You don't need the `&` in `&base_type`. This fails clippy.

Comment (BearLemma) [reply server/src/datastore/crud.rs]: fixed

**Review by espindola — APPROVED** [review]
